// @harv46/vue-table/dist/es.js
import { defineComponent as i, openBlock as r, createElementBlock as a, createElementVNode as o, normalizeClass as n, Fragment as d, renderList as u, toDisplayString as c, renderSlot as k } from "vue";
const h = { class: "relative overflow-x-auto shadow-md sm:rounded-lg" }, m = { key: 0 }, f = { key: 1 }, _ = /* @__PURE__ */ i({
  __name: "VueTable",
  props: {
    headers: {},
    data: {},
    keys: {},
    dark: { type: Boolean, default: !1 }
  },
  setup(y) {
    const g = (e, p) => p.reduce((t, s) => t == null ? void 0 : t[s], e);
    return (e, p) => (r(), a("div", h, [
      o("table", {
        class: n(["w-full text-sm text-left text-gray-500", { "dark:text-gray-400": e.dark }])
      }, [
        o("thead", {
          class: n(["text-xs text-gray-700 uppercase bg-gray-50", { "dark:bg-gray-700 dark:text-gray-400": e.dark }])
        }, [
          o("tr", null, [
            (r(!0), a(d, null, u(e.headers, (t, s) => (r(), a("th", {
              key: s,
              scope: "col",
              class: "px-6 py-3"
            }, c(t), 1))), 128)),
            k(e.$slots, "th")
          ])
        ], 2),
        o("tbody", null, [
          (r(!0), a(d, null, u(e.data, (t, s) => (r(), a("tr", {
            class: n(["bg-white border-b hover:bg-gray-50", {
              "dark:bg-gray-800 dark:border-gray-700 dark:hover:bg-gray-600": e.dark
            }]),
            key: s
          }, [
            (r(!0), a(d, null, u(e.keys, (l, b) => (r(), a("td", {
              key: b,
              class: "px-6 py-4"
            }, [
              typeof l == "object" ? (r(), a("span", m, c(g(t, l)), 1)) : (r(), a("span", f, c(t[l]), 1))
            ]))), 128)),
            k(e.$slots, "td", { item: t })
          ], 2))), 128))
        ])
      ], 2)
    ]));
  }
});
const x = {
  install(y, g) {
    y.component("VueTable", _);
  }
};
export {
  _ as VueTable,
  x as default
};
